.github/workflows/docs.yml

```diff
@@ -0,0 +1,28 @@
+name: Generate and publish docs
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    name: Generate docs with jazzy and publish to Github pages
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Generate Docs
+        run: |
+          brew install sourcekitten
+          gem install jazzy
```

Comment:
Should this be added to the gemfile?

Reply:
(Maybe not, since it's only for doc generation? But would be good to stabilize the version somehow)

Reply:
Sure, that's easy enough.